### PR TITLE
[BugFix] fixbug when keyType in map is char or varchar for parquet

### LINKED ISSUE #12881: [HIVE-MAP] query error from hive external table with type: map<char(10),smallint> and parquet file format
### Steps to reproduce the behavior (Required)
1. create hive table stored as parquet and insert data:
```
create table hdfs_parquet_n(
col_int int,
col_map map<char(10),smallint> 
)
row format delimited fields terminated by ':'
stored as parquet
tblproperties("parquet.compression"="snappy");

insert into hdfs_parquet_n select 7,map(cast("abc-123-00" as char(10)),smallint(-327));
```

2. select catalog.hivedb.hivetbl in sr:
```
mysql> select * from hdfs_parquet_n;
ERROR 1064 (HY000): key in map group must be required:file = hdfs://emr-header-1.cluster-49091:9000/user/hive/warehouse/hive_extbl_test_struct_map.db/hdfs_parquet_n/000000_0
```

### Expected behavior (Required)
return the right result

### Real behavior (Required)
error

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> select current_version();
+----------------------+
| current_version()    |
+----------------------+
| MAIN-RELEASE 100ad04 |
+----------------------+
1 row in set (0.00 sec)
```

### PR DESCRIPTION
**Pull request by zombee0**
When keyType in map is char or varchar in hive, for parquet files it's key_schema is BYTE_ARRAY which is optional.

截屏2022-11-02 21 26 42

Signed-off-by: zombee0 <flylucas_10@163.com>

## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #12881 

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: be/src/formats/parquet/schema.cpp [hunk 1]
```diff
@@ -190,7 +190,9 @@ Status SchemaDescriptor::map_to_field(const std::vector<tparquet::SchemaElement>
         return Status::InvalidArgument("key_value in map group must be a repeated group");
     }
     auto& key_schema = t_schemas[pos + 2];
-    if (!is_required(key_schema)) {
+    // when key type is char or varchar in hive, not string
+    // the real type is BYTE_ARRAY which is OPTIONAL
+    if ((!is_required(key_schema)) & (key_schema.type != tparquet::Type::type::BYTE_ARRAY)) {
```
Comment (DorianZheng) [inline be/src/formats/parquet/schema.cpp:195]: Is it documented, seems it's not match with LOC:173

Comment (DorianZheng) [reply be/src/formats/parquet/schema.cpp:195]: https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#maps

Comment (Smith-Cruise) [inline be/src/formats/parquet/schema.cpp:195]: `&&` ?

**Review by DorianZheng — DISMISSED** [review]
LGTM

File: be/src/formats/parquet/schema.cpp (hunk 1 above)
Comment (zombee0) [reply be/src/formats/parquet/schema.cpp:195]: when both & and && work,  & is better.

Comment (Smith-Cruise) [reply be/src/formats/parquet/schema.cpp:195]: I think that's a problem from Hive, Hive does not set the key as `required` when the key is BYTE_ARRAY.

Do I understand right?

**Review by Smith-Cruise — DISMISSED** [review]

**Review by Youngwb — DISMISSED** [review]

**Review by dirtysalt — DISMISSED** [review]

**Comment by dirtysalt** [comment]
run starrocks_admit_test

**Comment by zombee0** [comment]
run starrocks_admit_test

**Comment by zombee0** [comment]
run starrocks_admit_test

File: be/src/formats/parquet/schema.cpp (hunk 1 above)
Comment (imay) [inline be/src/formats/parquet/schema.cpp:195]:
```suggestion
    if ((!is_required(key_schema)) && (key_schema.type != tparquet::Type::type::BYTE_ARRAY)) {
```

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by zombee0** [comment]
run starrocks_admit_test

**Pull request merged**